pkg/updatestatus/nodeinformer.go

```diff
@@ -0,0 +1,450 @@
+		// The following condition is to handle the multi-arch migration because the version number stays the same there
+		(!ok || node.Annotations[mco.CurrentMachineConfigAnnotationKey] == desiredConfig)
+
+	// foundCurrent makes sure we don't blip phase "updating" for nodes that we are not sure
+	// of their actual phase, even though the conservative assumption is that the node is
+	// at least updating or is updated.
+	isUpdating := !isUpdated && foundCurrent && foundDesired && isLatestUpdateHistoryVersionEqualTo(cv.Status.History, desiredVersion)
+
+	condition := metav1.Condition{
+		Type:               string(NodeStatusInsightUpdating),
+		Status:             metav1.ConditionUnknown,
+		Reason:             string(NodeCannotDetermine),
+		LastTransitionTime: now,
+	}
+
+	phase := calculatePhase(mcp, node, isUpdating, isUpdated)
+	var estimate *metav1.Duration
+	switch phase {
+	case phaseStatePaused:
+		estimate = toPointer(time.Duration(0))
+		condition.Type = string(NodeStatusInsightUpdating)
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = string(NodePaused)
+	case phaseStatePending:
+		condition.Type = string(NodeStatusInsightUpdating)
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = string(NodeUpdatePending)
+	case phaseStateDraining:
+		estimate = toPointer(time.Duration(10 * time.Minute))
+		condition.Type = string(NodeStatusInsightUpdating)
+		condition.Status = metav1.ConditionTrue
+		condition.Reason = string(NodeDraining)
+	case phaseStateUpdating:
+		estimate = toPointer(time.Duration(10 * time.Minute))
+		condition.Type = string(NodeStatusInsightUpdating)
+		condition.Status = metav1.ConditionTrue
+		condition.Reason = string(NodeUpdating)
+	case phaseStateRebooting:
+		estimate = toPointer(time.Duration(10 * time.Minute))
+		condition.Type = string(NodeStatusInsightUpdating)
+		condition.Status = metav1.ConditionTrue
+		condition.Reason = string(NodeRebooting)
+	case phaseStateUpdated:
+		estimate = toPointer(time.Duration(0))
+		condition.Type = string(NodeStatusInsightUpdating)
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = string(NodeCompleted)
+	}
+
+	if isUnavailable && !isUpdating {
+		estimate = nil
+		if isUpdated {
+			estimate = toPointer(time.Duration(0))
+		}
+		condition.Type = string(NodeStatusInsightAvailable)
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = lns.GetUnavailableReason()
+		condition.Message = lns.GetUnavailableMessage()
+		condition.LastTransitionTime = metav1.Time{Time: lns.GetUnavailableSince()}
+	}
+
+	if isDegraded {
+		estimate = nil
+		if isUpdated {
+			estimate = toPointer(time.Duration(0))
+		}
+		condition.Type = string(NodeStatusInsightDegraded)
+		condition.Status = metav1.ConditionTrue
+		condition.Reason = node.Annotations[mco.MachineConfigDaemonReasonAnnotationKey]
+		condition.Message = node.Annotations[mco.MachineConfigDaemonReasonAnnotationKey]
+	}
```

Comment:
This does not look correct. There should always be three conditions in the insight: `Updating`, `Degraded`, `Unavailable`. The blocks here always overwrite a single `condition` structure, but the code should manage all three (including things like setting `Degraded` to `False` when it is not degraded.

Reply:
Done.